### PR TITLE
pio-package: add page

### PR DESCRIPTION
**Pull request by marchersimon**
For #5517

### CODE REVIEW

File: pages/common/pio-package.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# pio package
+
+> Manage packages in the registry.
+> Packages can only be removed within 72 hours from the published date.
```
Comment (bl-ue) [inline pages/common/pio-package.md:4]:
```suggestion
> Packages can only be removed within 72 hours (3 days) from the published date.
```

File: pages/common/pio-package.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+> More information: <https://docs.platformio.org/en/latest/core/userguide/package/index.html>.
```
Comment (bl-ue) [inline pages/common/pio-package.md:5]:
```suggestion
> More information: <https://docs.platformio.org/en/latest/core/userguide/package/>.
```

File: pages/common/pio-package.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+- Remove all versions of a published package from the registry:
+
+`pio package unpublish {{package_name}}`
+
+- Remove a specific version of a published package from the registry:
+
+`pio package unpublish {{package_name}}@{{version}}`
+
+- Undo the removal, putting all or a specific version of the package back into the registry:
```
Comment (bl-ue) [inline pages/common/pio-package.md:35]:
```suggestion
- Undo the removal, putting all versions or a specific version of the package back into the registry:
```

File: pages/common/pio-package.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# pio package
+
+> Manage packages in the registry.
+> Packages can only be removed within 72 hours from the published date.
+> More information: <https://docs.platformio.org/en/latest/core/userguide/package/index.html>.
+
+- Create a tarball from the current package directory:
```
Comment (bl-ue) [inline pages/common/pio-package.md:7]:
```suggestion
- Create a tarball from the current directory:
```

File: pages/common/pio-package.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+`pio package pack --output {{path/to/output_tarball}}`
+
+- Publish the current package directory:
```
Comment (bl-ue) [inline pages/common/pio-package.md:11]:
```suggestion
- Publish the current directory:
```

File: pages/common/pio-package.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+`pio package publish`
+
+- Publish the current package directory and restrict public access to it:
```
Comment (bl-ue) [inline pages/common/pio-package.md:15]:
```suggestion
- Publish the current directory and restrict public access to it:
```

File: pages/common/pio-package.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+`pio package publish --private`
+
+- Publish a specific package:
+
+`pio package publish {{path/to/package}}`
```
Comment (bl-ue) [inline pages/common/pio-package.md:21]:
```suggestion
`pio package publish {{path/to/package.tar.gz}}`
```

File: pages/common/pio-package.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# pio package
+
+> Manage packages in the registry.
+> Packages can only be removed within 72 hours from the published date.
+> More information: <https://docs.platformio.org/en/latest/core/userguide/package/index.html>.
+
+- Create a tarball from the current package directory:
+
+`pio package pack --output {{path/to/output_tarball}}`
```
Comment (bl-ue) [inline pages/common/pio-package.md:9]:
```suggestion
`pio package pack --output {{path/to/output_tarball.tar.gz}}`
```

File: pages/common/pio-package.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+`pio package publish --private`
+
+- Publish a specific package:
+
+`pio package publish {{path/to/package}}`
+
+- Publish a specific package and set custom release date and time (UTC):
+
+`pio package publish {{path/to/package}} --released-at "{{2021-04-08 21:15:38}}"`
```
Comment (bl-ue) [inline pages/common/pio-package.md:25]:
```suggestion
`pio package publish {{path/to/package.tar.gz}} --released-at "{{2021-04-08 21:15:38}}"`
```

**Review by bl-ue — COMMENTED** [review]
See my suggestions.  The valid extensions are `.tar.gz` and `.zip`; I picked `.tar.gz` because it's more Unix oriented and is traditionally what's meant by 'tarball'.  I think it's good to have extensions on the tarball paths though because it makes it clear that they're to be files and not folders.

Also, I removed 'package' from 'current package directory' because that looks just a tad bit odd to me, and I think it conveys the same meaning without it; however, you can ignore that change if you want to.

File: pages/common/pio-package.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
+# pio package
+
+> Manage packages in the registry.
+> Packages can only be removed within 72 hours (3 days) from the published date.
```
Comment (bl-ue) [inline pages/common/pio-package.md:4]:
```suggestion
> Packages can only be removed within 72 hours (3 days) from the date they're published.
```

File: pages/common/pio-package.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+> More information: <https://docs.platformio.org/en/latest/core/userguide/package/>.
+
+- Create a tarball from the current directory:
```
Comment (bl-ue) [inline pages/common/pio-package.md:7]:
```suggestion
- Create a package tarball from the current directory:
```

File: pages/common/pio-package.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+`pio package pack --output {{path/to/output_tarball.tar.gz}}`
+
+- Publish the current directory:
```
Comment (bl-ue) [inline pages/common/pio-package.md:11]:
```suggestion
- Create and publish a package tarball from the current directory:
```

File: pages/common/pio-package.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+- Create and publish a package tarball from the current directory:
+
+`pio package publish`
+
+- Publish the current directory and restrict public access to it:
+
+`pio package publish --private`
+
+- Publish a specific package:
```
Comment (bl-ue) [inline pages/common/pio-package.md:19]:
```suggestion
- Publish a package:
```

File: pages/common/pio-package.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+`pio package publish {{path/to/package.tar.gz}}`
+
+- Publish a specific package and set custom release date and time (UTC):
```
Comment (bl-ue) [inline pages/common/pio-package.md:23]:
```suggestion
- Publish a package with a custom release date (UTC):
```

**Comment by marchersimon** [comment]
> The valid extensions are .tar.gz and .zip; I picked .tar.gz because it's more Unix oriented and is traditionally what's meant by 'tarball'. I think it's good to have extensions on the tarball paths though because it makes it clear that they're to be files and not folders.

Actually I think that you can either specify a file or a directory.

> pio package publish [<source directory, tar.gz or zip>] [OPTIONS]

**Review by bl-ue — APPROVED** [review]

File: pages/common/pio-package.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
+# pio package
+
+> Manage packages in the registry.
+> Packages can only be removed within 72 hours (3 days) from the date that they are published.
+> More information: <https://docs.platformio.org/en/latest/core/userguide/package/>.
+
+- Create a package tarball from the current directory:
+
+`pio package pack --output {{path/to/output_tarball.tar.gz}}`
```
Comment (navarroaxel) [inline pages/common/pio-package.md:9]: The command argument is named `--output` and `tarball` doesn't add a meaning to the output file. I think `package.tar.gz` or maybe `my_package.tar.gz`
```suggestion
`pio package pack --output {{path/to/package.tar.gz}}`
```

Comment (bl-ue) [reply pages/common/pio-package.md:9]: I like the former, `package.tar.gz`.

**Comment by bl-ue** [comment]
>Actually I think that you can either specify a file or a directory.

Oh, I didn't notice that message.  Well, I doubt that it matters to document that...but idk.

**Comment by marchersimon** [comment]
I'd say that's what the `More information` link is for.

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**